File: poo/constructor/class.py
class Employee():

    def __init__(self, id, name):
        self.id = id
        self.name = name

    def set_id(self, id):
        self.id = id
    
    def set_name(self, name):
        self.name = name
    
    def get_id(self):
        return self.id
    
    def get_name(self):
        return self.name

employee = Employee(0, "Ana")
employee.set_id(1)
employee.set_name("Maria")

print(employee.get_id())
print(employee.get_name())

File: poo/constructor/class.go
package main

import "fmt"

type Employee struct {
	id       int
	name     string
	vacation bool
}

func NewEmployee(id int, name string, vacation bool) *Employee {
	return &Employee{
		id:       id,
		name:     name,
		vacation: vacation,
	}
}

func main() {
	e := Employee{}
	fmt.Printf("%v\n", e)

	e2 := Employee{
		id:       1,
		name:     "Maria",
		vacation: true,
	}
	fmt.Printf("%v\n", e2)

	e3 := new(Employee)
	fmt.Printf("%v\n", *e3)
	e3.id = 1
	fmt.Printf("%v\n", *e3)

	// 4
	e4 := NewEmployee(1, "Name", false)
	fmt.Printf("%v\n", *e4)

}
